### DISCUSSION TITLE
The current project could not be installed: No file/folder found for package

### DISCUSSION THREAD

**Discussion by NotArobase** [python-poetry/poetry#9835] [Q&A]
Hello, i'm currently using an open source project that used poetry and i'm having quite the problem.

When using poetry install, i get the error : 

```
Warning: The current project could not be installed: No file/folder found for package myproject
If you do not want to install the current project use --no-root.
If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
In a future version of Poetry this warning will become an error!
```

```
Myproject
├──package1
│  ├──__init__.py
│  └──something.py
├──package2
│  ├──__init__.py
│  └──something.py
├──pyproject.toml
└──README.rd
```

Inside my pyproject.toml i got : 

```
[tool.poetry]
name = "Myproject"
```

why i try to run tests, it doesn't found the package package1 and package2. 

I did everything inside a peotry shell.

If you have any idea why, let me know ! 

Thanks

**Comment by finswimmer** [comment]
Hey @NotArobase,

if you don't explicit tell Poetry where your packages are located (https://python-poetry.org/docs/pyproject/#packages), Poetry assumes that there is a folder with the same name as given in `name` in the `pyproject.toml` either at the same level as the `pyproject.toml` or within a folder called `src`.

So what you probably want is:

```
[tool.poetry]
name = "Myproject"
packages = [
    { include = "package1" },
    { include = "package2 },
]
```

Once you've run `poetry install` once again, you should be able to do `import package1` and `import package2`.

I personal would recommend, that `packag1` and `package2` should share the same namespace. So your imports would look like `import myproject.package1` and `import myproject.package2`.

For this your folder structure must look like this:

```
Myproject
├── README.md
├── myproject
│   ├── __init__.py
│   ├── package1
│   │   └── __init__.py
│   └── package2
│       └── __init__.py
└── pyproject.toml
```

No changes are then require to your `pyproject.toml`.

fin swimmer

**Reply by cwhead-globus** [reply]
Is there a way this can be done using `[project]` rather than `[tool.poetry]`?

**Reply by Secrus** [reply]
`packages` is a `[tool.poetry]` directive, so you have to use it, but the `name` part can be done via `[project]`
